### DISCUSSION TITLE
When Hears does not return tweet information data

### DISCUSSION THREAD

**Discussion by spartyboy** [spatie/laravel-twitter-streaming-api#45] [Ideas]
when i use when hear this is what it returns i have no access to the tweet object itself

{"data":{"id":"1565749696433463297","text":"@spartacusyusuf hss"},"matching_rules":[{"id":"1565749486877642753","tag":"@spartacusyusuf"}]}

**Comment by freekmurze** [comment]
Feel free to PR improvements.

**Reply by spartyboy** [reply]
But according to docs we have access to the tweet object.

**Reply by spartyboy** [reply]
But when I try accessing the array to get the user and the tweet information it shows undefined

**Reply by spartyboy** [reply]
i was able to fix this by doing the following please do update documentation

`
  TwitterStreamingApi::publicStream()
        ->whenHears(['@spartacusyusuf','@YusufAb24773901'], function(array $tweet) {
            echo json_encode($tweet)."\n";
        })
        ->startListening(new Sets(new Fieldset('tweet.fields','public_metrics,conversation_id,entities'),new Fieldset('user.fields','id,name,username,profile_image_url,verified,public_metrics')));
`
adding the Sets to the start listening method and passing the fieldset in the Sets Contructor to define the fields to append to the stream ed data is the solution if you don't get the full tweet object. this package depends on twitter version 2 api  and was documented here [Twitter Stream](https://developer.twitter.com/en/docs/tutorials/stream-tweets-in-real-time)
